untracked inputs in constraints

avikchaudhuri opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109037

Differential Revision: [D49157009](https://our.internmc.facebook.com/intern/diff/D49157009/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

avikchaudhuri committed:
untracked inputs in constraints



Differential Revision: [D49157009](https://our.internmc.facebook.com/intern/diff/D49157009/)

[ghstack-poisoned]

zhxchen17 commented on torch/_export/exported_program.py:
```diff
@@ -337,7 +337,7 @@ def _process_constraints(
         assert isinstance(val, FakeTensor)
         symint = val.shape[input_dim.dim]
-        assert isinstance(symint, SymInt)
+        assert isinstance(symint, SymInt), f"Expected SymInt but got {symint}: {type(symint)}"
```
@avikchaudhuri does this mean we still ran into assertion failure?

avikchaudhuri replied:
Only if you try passing a 0 or 1 dimensional tensor in KJT, yes. In all other cases this assert should not fail.

avikchaudhuri replied:
For KJT you get an int that is 0 or 1.

zhxchen17 replied:
ah I see, it makes sense now.

avikchaudhuri committed:
Update on "untracked inputs in constraints"


Differential Revision: [D49157009](https://our.internmc.facebook.com/intern/diff/D49157009/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

avikchaudhuri commented:
@pytorchbot merge

zhxchen17 commented:
@pytorchbot merge

avikchaudhuri committed:
Update on "untracked inputs in constraints"


Differential Revision: [D49157009](https://our.internmc.facebook.com/intern/diff/D49157009/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

avikchaudhuri commented:
@pytorchbot merge -f "failing test no longer exists in trunk, see https://github.com/pytorch/pytorch/pull/108750"
